### ISSUE TITLE
import() fails when you have a variable string importing a Vue component

### ISSUE THREAD

**Issue by pikax** [vitest-dev/vitest#813]
### Describe the bug

When you use `import()` if the string is a variable it doesn't work, if is a variable it must have an absolute path.

```ts
// tests/test.spec.ts
import { test, expect } from "vitest";

test("test import", async () => {
  const Hello = import("./../src/components/HelloWorld.vue");

  const HelloPath = "./../src/components/HelloWorld.vue";
  const HelloOther = import(HelloPath); // fails here

  expect(Hello).toBe(HelloOther);
});
```

From my findings the request fails here [directRequest](https://github.com/vitest-dev/vitest/blob/5ef622aabfa94b953d7870a9203d6429ce0f5106/packages/vite-node/src/client.ts#L74-L75), the `id` seems to still be the string provided on the import, when you pass a static string it will convert the relative path into the absolute.

### Reproduction

[StackBlitz](https://stackblitz.com/edit/vitest-dev-vitest-trk5mv?file=test/basic.test.ts)

```ts
// basic.test.ts
import { test, expect } from 'vitest';

test('test import', async () => {
  const Hello = await import('./../src/components/HelloWorld.vue');

  const HelloPath = './../src/components/HelloWorld.vue';
  const HelloOther = await import(HelloPath); // fails here

  expect(Hello).toBe(HelloOther);
});
```

### System Info

```shell
System:
    OS: Windows 10 10.0.22000
    CPU: (24) x64 AMD Ryzen 9 3900X 12-Core Processor
    Memory: 23.50 GB / 63.94 GB
  Binaries:
    Node: 16.10.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.13 - C:\Program Files\nodejs\yarn.CMD
    npm: 7.24.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22000.120.0), Chromium (98.0.1108.56)
    Internet Explorer: 11.0.22000.120
  npmPackages:
    @vitejs/plugin-vue: ^2.2.2 => 2.2.2
    vite: ^2.8.4 => 2.8.4
    vitest: ^0.4.1 => 0.4.
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by pikax** [comment]
@antfu is this released? Still having the issue on `0.5.0` [StackBlitz](https://stackblitz.com/edit/vitest-dev-vitest-trk5mv?file=test/basic.test.ts) still failing 🤔

**Comment by sheremet-va** [comment]
> @antfu is this released? Still having the issue on `0.5.0` [StackBlitz](https://stackblitz.com/edit/vitest-dev-vitest-trk5mv?file=test/basic.test.ts) still failing 🤔

It failes because they are not equeal. That is another issue. Importing works fine
